src/test/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloudIT.java

```diff
@@ -362,6 +367,51 @@ public void testTemplateNoGoogleLabels() throws Exception {
         }
     }
 
+    // Tests that no snapshot is created when we only have successful builds for given node
+    @Test(timeout = 300000)
+    public void testNoSnapshotCreated() throws Exception {
+        logOutput.reset();
+
+        ComputeEngineCloud cloud = (ComputeEngineCloud) r.jenkins.clouds.get(0);
+        cloud.addConfiguration(validInstanceConfiguration());
+
+        FreeStyleProject project = r.createFreeStyleProject();
+        Builder step = new Shell("echo works");
+        project.getBuildersList().add(step);
+        project.setAssignedLabel(new LabelAtom(LABEL));
+
+        FreeStyleBuild build = r.buildAndAssertSuccess(project);
+        Node worker = build.getBuiltOn();
+        worker.toComputer().doDoDelete();
+
+        assertFalse(logs(), logs().contains("snapshot"));
```

Comment:
This seems like a tenuous approach for asserting this. Consider modeling the approach used in testSnapshotCreated(), but instead asserting the negative.

Reply:
I tried to get a snapshot with the node name, but that leads to an exception (trying to get a snapshot that doesn't exist).

Reply:
Not a problem, you can use the thrown exception as the expected case, see: https://www.baeldung.com/junit-assert-exception

Reply:
done

Reply:
For me that assertion dont make much sense..

If we are checking if we dont have snapshot that is all what we need..
And what it means that there is no `snapshot` in log?

Reply:
Because when it begins to delete, I have a log statement that says snapshot is being created. 
The assertion is because the result of the get for a non-existent snapshot is not null; instead, it leads to a GoogleJsonResponseException.

Reply:
If the delete is what throws the exception, then the assertion will never be executed. Suggest removing the dead code after the exception source.

Reply:
The assertion is for a log statement, so it does get executed.